_build-deps.yml

```diff
@@ -13,3 +13,35 @@ steps:
 - bash: |
     ./install-crosstool-ng.sh
   displayName: 'Build and install crosstool-ng'
+
+- bash: |
+    set -e
+    git fetch --tags
+
+    # git-describe --always almost does what we want, but we need to connect the
+    # ReleaseTag variable contents to how this build was triggered, which we
+    # will find out using `Build.SourceBranch` (a pipeline varaible).
+    #
+    # Importantly, a few things are going on here:
+    # - If we were triggered by a tag, we need to set ReleaseTag to exactly the
+    #   tag name, so that the artifacts are named correctly. If we cannot do
+    #   this, the build needs to fail rather than uploading artifacts to some
+    #   other random tag.
+    # - If we were triggered by a branch build, we need to be more careful, as
+    #   tagged commits are also pushed to branches. So branch builds explicitly
+    #   use the longer format so if that commit matches a tag, the ReleaseTag is
+    #   not a tag name.
```

Comment:
Nit: the content is fine but this last sentence came out a little weird in its form. Consider tweaking.